### PR TITLE
LPD-28713 Required permission error after assign workflow task

### PR DESCRIPTION
**Pull request by victorKammerer**
[LPD-28713](https://liferay.atlassian.net/browse/LPD-28713)

[LPD-28713]: https://liferay.atlassian.net/browse/LPD-28713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rijema** [comment]
ci:test:sf

**Comment by rijema** [comment]
Hi @victorKammerer , could you please take a look at the SF failing report? Maybe it's necessary to change something at review-user-notification-workflow-definition.xml . Thanks!

**Comment by igor-franca** [comment]
Hey @victorKammerer and @rijema, this SF error already be fixed in Carlos's PR.  
Also I perform tests in c68b2aaff2f035537c90f418195775dcd18414d4 and according to the tests carried out, everything is fine with the solution, the remaining tests commits will be reviewed as soon as possible.

CC: @linolaoj

File: modules/test/playwright/pages/roles-admin-web/RolesPage.ts [hunk 1]
```diff
@@ -11,14 +11,31 @@ export class RolesPage {
 	readonly applicationsMenuPage: ApplicationsMenuPage;
 	readonly page: Page;
 	readonly userLink: Locator;
+	readonly selectRoleButton: Locator;
+	readonly optionsButton: Locator;
+	readonly deleteButton: Locator;
```
Comment (igor-franca) [inline modules/test/playwright/pages/roles-admin-web/RolesPage.ts:16]: Hey Victor, could you sort this locator?

File: modules/test/playwright/pages/roles-admin-web/RolesPage.ts [hunk 2]
```diff
@@ -11,14 +11,31 @@ export class RolesPage {
...
 
 	constructor(page: Page) {
 		this.applicationsMenuPage = new ApplicationsMenuPage(page);
 		this.page = page;
 		this.userLink = page.getByRole('link', {exact: true, name: 'User'});
+		this.optionsButton = page.getByLabel('Options', { exact: true });
+		this.deleteButton = page.getByRole('menuitem', { name: 'Delete' });
```
Comment (igor-franca) [inline modules/test/playwright/pages/roles-admin-web/RolesPage.ts:23]: Hey Victor, could you sort this?

File: modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts [hunk 3]
```diff
@@ -6,76 +6,158 @@
...
+
+	await workflowTasksPage.goToReviewPage(blogTitle);
+
+	await page.waitForTimeout(3000);
+
+	await workflowReviewTasksPage.clickReviewActionMenu();
+
+	await workflowReviewTasksPage.clickAssignToMenuItem();
+		
+	await workflowReviewTasksPage.selectAssignee(user2.id.toString());
```
Comment (igor-franca) [inline modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts:153]: Hey Victor, I can se the interface declaration of user:

```
type TUserAccount = {
	alternateName?: string;
	emailAddress?: string;
	familyName?: string;
	givenName?: string;
	id?: string;
	name?: string;
};
```

I noticed that this ID is already a string. Could you check in the API return if this property is actually a string? If it is a number, I believe we can correct this interface.

File: modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts [hunk 4]
```diff
@@ -6,76 +6,158 @@
...
+
+	await workflowReviewTasksPage.clickDoneAssigneeButton();
+
+	const user2Id = parseInt(user2.id);
```
Comment (igor-franca) [inline modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts:157]: By correcting the interface, we will not need this line if the return is a number.

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey @victorKammerer, nice work here!

I would like to suggest reorganizing your commits.

You could carefully squash your commits, so that snippets from your initial test creation phase such as:

`test('please work', async ({` :sweat_smile:

do not appear during the review. 

These initial commits are valid for the purpose of saving current progress in the remote environment. However, we can do some final work to polish these commits before sending the PR.

Furthermore, before performing a push, remember to run the source format, which should point out some formatting improvements.

**Comment by victorKammerer** [comment]
Hey, @igor-franca! All requested changes have been made!

File: modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts [hunk 5]
```diff
@@ -152,11 +144,14 @@ test('send user back to my workflow tasks page after assign another user to revi
...
+	const user2 =
+		await apiHelpers.headlessAdminUser.getUserAccountByEmailAddress(
+			'test@liferay.com'
+		);
 
-	const user2Id = parseInt(user2.id);
+	await workflowReviewTasksPage.selectAssignee(user2.id.toString());
 
-	await apiHelpers.headlessAdminUser.deleteUserAccount(user2Id);
+	await workflowReviewTasksPage.clickDoneAssigneeButton();
```
Comment (igor-franca) [inline modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts:156]: Hey, Victor. Can we add an additional step to the test by asserting that the user was directed to the desired page?

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey Victor, great work here. Just a few details

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by victorKammerer** [comment]
ci:test:playright

**Comment by igor-franca** [comment]
ci:test:playwright

File: modules/test/playwright/pages/roles-admin-web/RolesPage.ts
Comment (igor-franca) [inline modules/test/playwright/pages/roles-admin-web/RolesPage.ts:1]: Hey @victorKammerer, after your latest changes, I believe this page is no longer being used. Could you confirm me?

File: modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import {getRandomInt} from '../../utils/getRandomInt';
+import getRandomString from '../../utils/getRandomString';
+import performLogin, {performLogout} from '../../utils/performLogin';
+import {blogsPagesTest} from '../blogs-web/fixtures/blogsPagesTest';
+
+export const test = mergeTests(
+	apiHelpersTest,
+	blogsPagesTest,
+	loginTest(),
+	rolesPagesTest,
```
Comment (igor-franca) [inline modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts:22]: If it is really not being used, you might remember to remove its call here

File: modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+import {readFileSync} from 'fs';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {rolesPagesTest} from '../../fixtures/rolesPagesTest';
```
Comment (igor-franca) [inline modules/test/playwright/tests/portal-workflow-task-web/assignments.spec.ts:11]: If it is really not being used, you might remember to remove its import here

**Review by igor-franca — CHANGES_REQUESTED** [review]

**Comment by victorKammerer** [comment]
Done, @igor-franca!

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by victorKammerer** [comment]
ci:test:playwright

**Comment by linolaoj** [comment]
ci:test:sf

**Comment by linolaoj** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
Hey @victorKammerer, could you check the SF error:

`Error: Cannot find module './utils/getWorkflowDefinition'` and check if this wasn't something that Carlos renamed and after the rebase this problem occurred?

**Comment by victorKammerer** [comment]
Done, @igor-franca!

**Comment by linolaoj** [comment]
ci:test:palywright

**Comment by linolaoj** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
We identified a related failure in one test, we have already carried out an initial investigation and Victor is working on it.

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
ci:test:sf

**Review by igor-franca — CHANGES_REQUESTED** [review]
Good job Victor, however I believe that with the latest changes, the test that was failing has become intermittent. I added an instruction on how to proceed regarding the asynchronous code, and I believe that the problem will be solved with this modification.

Furthermore, I renamed the xml file used, passing a striking characteristic of the xml to the file name.

Finally, as a last detail. I reverted the position modification of the `BlogsPage` file that did not belong to the BPM component. Maybe this modification would make sense at the beginning of the implementation, but after some rebases. I believe that the reason for this change has been lost over time, as the ordering is no longer correct.

I am making a [branch](https://github.com/liferay/liferay-portal/compare/master...igor-franca:liferay-portal:LPP-54486-2) available with the suggested modifications, you can contact me with any questions

**Comment by victorKammerer** [comment]
Thank you for your suggestions @igor-franca, all done!

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by rijema** [comment]
ci:test:sf

**Comment by rijema** [comment]
ci:test:workflow

**Comment by igor-franca** [comment]
ci:report:35047676

**Comment by victorKammerer** [comment]
@igor-franca, I reproduced the errors on the UI, and it worked fine. So, I guess it may be unrelated to the changes at this branch.

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey @victorKammerer, I noticed that the `WorkflowReviewTaskPage.ts` and `WorkflowTaskDetailsPage.ts` files are very similar and seem to represent the same page. You can check if after conflict resolution you chose to keep one of the files when Carlos renamed it, thus creating a new file.

If this is indeed what happened, I would like to request that we only keep the` WorkflowTaskDetailsPage.ts` file by inserting your contributions in it and removing the `WorkflowReviewTaskPage.ts` file. If that really is the case.

Additionally, we can add a final assert to ensure that the user has been directed to the desired page. Thus removing the final wait.

As we will not be making changes to the application, we will only need to run the source format and playwright test after these changes.

**Comment by victorKammerer** [comment]
ci:test:sf

**Comment by igor-franca** [comment]
ci:test:sf

**Review by igor-franca — APPROVED** [review]
Nice Work @victorKammerer!

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
Hey @victorKammerer, could you check the playwright results?

**Comment by igor-franca** [comment]
ci:forward

**Pull request closed without merging**